lib/model/model.go

```diff
@@ -1486,7 +1541,7 @@ func (m *Model) AddConnection(conn connections.Connection, hello protocol.HelloR
 	conn.ClusterConfig(cm)
 
 	device, ok := m.cfg.Devices()[deviceID]
-	if ok && (device.Name == "" || m.cfg.Options().OverwriteRemoteDevNames) {
+	if ok && (device.Name == "" || m.cfg.Options().OverwriteRemoteDevNames) && hello.DeviceName != "" {
```

Comment:
This is where 4 hours were spent essentially.

We setup the model in newState, AddConnection kicks off m.cfg.SetDevice(), we start poking at the model ourselves, adding folders, configs bla bla, and both newState and our poking kicks off a ton of routines which call CommitConfiguration in random fucking order, which leads to random fucking behaviour.

Reply:
We could .Wait() on the SetDevice promise to fix this too, but the amount of places where in practice we should wait for stuff, is through the fucking roof.